corehq/apps/reports/tests/test_esaccessors.py

```diff
@@ -1140,15 +1107,15 @@ def test_group_search_query(self):
         )
 
 
-class TestCaseESAccessors(BaseESAccessorsTest):
-
-    es_index_infos = [CASE_INDEX_INFO, USER_INDEX_INFO]
+@es_test(requires=[case_adapter, user_adapter])
+class TestCaseESAccessors(TestCase):
```

Comment:
```suggestion
class TestCaseESAccessors(TestCase):

    domain = "test_domain"
```

This test using a UUID for the domain is a bit sneaky. This is only required by one of the tests (`test_get_case_types_caching()`) which cleverly takes advantage of the unique domain name in order to prevent cache poisoning by other[/previous] tests. To clarify this for the "next dev", I think it's better to only use the unique domain where it is _required_, and provide an explanation why it is needed. For example:

```python
    def test_get_case_types_caching(self):
        # Use a unique domain to prevent quickcache from being poisoned by other
        # (and/or previous) tests.
        domain = uuid.uuid4().hex
        self._send_case_to_es(domain=domain, case_type='t1')

        # replace 'self.domain' with 'domain' throughout the rest of the test ...
```

Any other tests that are disrupted by caching should be decorated with `@disable_quickcache` (I think the only two that need it are `test_get_case_types()` and `test_get_case_types_case_sensitive()`).

Reply:
Ideally, one test run should not corrupt the environment, I decided to clear the cache after every test run. See https://github.com/dimagi/commcare-hq/pull/32522/commits/65dfc34ce935e7851f99b7b75feca78ac1caef9f 
Further, I investigated and realized that we don't need a new group index and user index in every test. So moving them to `setupClass` felt like a good way to go. See https://github.com/dimagi/commcare-hq/pull/32522/commits/3220da500563a92eee5f51a5dd3c8cd81e02c2e8 This resulted in saving around 15 seconds.